Maksss-ssskaM/labor/exchange

Maksss-ssskaM opened the pull request.

Maksss-ssskaM committed:
Jobs
- missing fields were added to models\jobs.py
- a schema was created for the job class

Maksss-ssskaM committed:
Responses
- missing fields were added to models\responses.py
- a schema was created for the response class

Maksss-ssskaM committed:
Jobs queries
- created queries for the job controller
- created stubs for the service responsible for business logic

Maksss-ssskaM committed:
Merge pull request #1 from Maksss-ssskaM/feat/responses

Responses

Maksss-ssskaM committed:
Merge pull request #2 from Maksss-ssskaM/feat/jobs

Feat/jobs

Maksss-ssskaM committed:
Jobs queries
- Business logic in the "Job" service is implemented

Maksss-ssskaM committed:
Merge pull request #3 from Maksss-ssskaM/feat/jobs

Jobs queries

Maksss-ssskaM committed:
Jobs routers and queries
- Added a deletion router
- Written business logic for it

Maksss-ssskaM committed:
Merge pull request #4 from Maksss-ssskaM/feat/jobs

Jobs routers and queries

Maksss-ssskaM committed:
Responses routers and queries
- Business logic in the "Response" service is implemented
- Created queries for the job controller
- Fixed a bug in UserSchema

Maksss-ssskaM committed:
Merge pull request #5 from Maksss-ssskaM/feat/responses

Responses routers and queries

Maksss-ssskaM committed:
Changed the name in one of the functions

Maksss-ssskaM committed:
Merge pull request #6 from Maksss-ssskaM/feat/jobs

Changed the name in one of the functions

Maksss-ssskaM committed:
Added query validation
- Added validation for company
- Improved typing

Maksss-ssskaM committed:
Merge pull request #7 from Maksss-ssskaM/feat/jobs

Added query validation

Maksss-ssskaM committed:
Added query validation
- Added validation to re-create a response
- Improved typing

Maksss-ssskaM committed:
Merge branch 'main' of github.com:Maksss-ssskaM/labor_exchange into feat/responses

Maksss-ssskaM committed:
Merge pull request #8 from Maksss-ssskaM/feat/responses

Added query validation

Maksss-ssskaM committed:
Added query validation
- Added validation of user re-creation

Maksss-ssskaM committed:
Merge pull request #9 from Maksss-ssskaM/feat/users

Feat/users

Maksss-ssskaM committed:
Added Unit Tests
-Added Unit tests for queries, routers and schemas
-Removed bugs

Maksss-ssskaM committed:
Merge pull request #10 from Maksss-ssskaM/feat/unit_tests

Added Unit Tests

Maksss-ssskaM committed:
Create python-package.yml

Maksss-ssskaM committed:
Git actions

Maksss-ssskaM committed:
Merge pull request #12 from Maksss-ssskaM/actions

Git actions

Rognak commented on src/migrations/versions/b6ccda8b1596_были_созданы_поля_в_модели_jobs_и_.py:
```diff
@@ -0,0 +1,40 @@
+"""Были созданы поля в модели jobs и responses
```
Всё хорошо. Вообще лучшей практикой считается одна модель - одна миграция. Тогда не будет каши в миграциях.

Rognak commented on src/queries/job.py:
Очень хорошо!

Rognak commented on src/queries/response.py:
```diff
@@ -0,0 +1,72 @@
+
+async def create_response(db: AsyncSession, response_schema: ResponseInSchema,
+                          current_user: User) -> Response | HTTPException:
+    job = await get_job_by_id(db=db, job_id=response_schema.job_id)
```
Лучше джобу передавать как зависимость, чтобы не было нарушения SOLID.
В данном случае нарушается S (Single Responsibility).

Rognak commented on src/queries/response.py:
```diff
@@ -0,0 +1,72 @@
+    elif current_user.is_company:
+        raise HTTPException(status.HTTP_403_FORBIDDEN, detail="Компании не могут оставлять отклик")
+
+    existing_response = await get_response_by_user_and_job_id(db=db, user_id=current_user.id,
```
Это можно сделать либо так как ты, либо обратиться к полю responses у конкретной джобы. Тогда ты сможешь посмотреть есть ли там юзер.

Rognak commented on src/queries/response.py:
```diff
@@ -0,0 +1,72 @@
+    return response
+
+
+async def get_responses_by_job_id(db: AsyncSession, job_id: int) -> List[Response]:
```
См. коммент выше. Можно сделать через обычную хелпер функцию.

Rognak commented on src/routers/job.py:
```diff
@@ -0,0 +1,86 @@
+        current_user: User = Depends(get_current_user)
+):
+    """
+    Create a new job with the provided details.
```
Документацию лучше писать на русском

Rognak commented on src/schemas/job.py:
```diff
@@ -0,0 +1,72 @@
+    return salary
+
+
+def validate_salary_to_not_less_than_from(salary, values, **kwargs):
```
Зачем вынес за схему?

Rognak commented on src/tests/routers/test_jobs.py:
```diff
@@ -0,0 +1,150 @@
+import random
+
+
+def generate_random_salary_range():
```
лучше вынести в какой-нибудь модель utils

Rognak commented on src/tests/schemas/test_jobs.py:
Отлично!

Rognak reviewed (commented):
Проверил. См. комментарии

Maksss-ssskaM committed:
db_problem

Maksss-ssskaM committed:
Deployed React app
-authorization
-components responsible for updating user data
-components for creating jobs
-components for creating responses to jobs

Maksss-ssskaM committed:
Update python-package.yml

Maksss-ssskaM committed:
Merge pull request #13 from Maksss-ssskaM/db_problem

Frontend
